### ISSUE TITLE
Time for a new release?

### ISSUE THREAD

**Issue by ppittle** [nuproj/nuproj#240]
Last release was ~9 months ago in Sept 2015.  There's been some great progress made since then and a good number of bug fixes.

Time for a new release? 

Not going to lie, hoping I can run an "offical" build rather than needing to build from source and distribute to my team.

Thanks!

**Comment by AArnott** [comment]
You can get the latest version of NuProj and NuProj.common nuget packages from here:
https://www.myget.org/F/aarnott/api/v3/index.json

You can also download the latest MSI from [AppVeyor's artifacts](https://ci.appveyor.com/project/terrajobst/nuproj/build/artifacts).

But I agree, ideally @terrajobst could either publish an update or grant more contributors the rights necessary to update it.

**Comment by ppittle** [comment]
Thanks, for the links.  Are you only hosting the NuGet projects on MyGet?  Or the VSIX as well?

I'm running MyGet as well and it'd be great to be able to mirror from yours.

**Comment by AArnott** [comment]
That's a great idea (to upload the VSIX to myget). I just tried, and it rejected it.

> It seems something went wrong...
The package "267c66da-0c67-4933-9ae7-a20c53608e49 0.10.58.50820" already exists. Overwriting existing packages is forbidden according to the package retention settings for this feed.

I don't know why -- I haven't ever uploaded a VSIX before.

**Comment by ppittle** [comment]
Do you have a space in your package ID?  I've had that problem before and MyGet rejected it.

**Comment by AArnott** [comment]
The problem is evidently that each VSIX (that targets VS2013, VS2015, and VS "15") has the same ID, and version. So that's the problem. I'm preparing a change to fix that.

**Comment by AArnott** [comment]
OK. I have the latest VSIX packages uploaded to the myget.org feed I mentioned earlier.

**Comment by petero-dk** [comment]
Hi @AArnott 

I was just trying your new feed, I have two issues with that:

1. It lists three packages with no noticeable distinction, perhaps adding the Visual Studio version in parenthesis could help (I had to guess it will fail on the wrong one)

2. Is is still not able to install the vsix because of the following error:

```
Install Error : Microsoft.VisualStudio.ExtensionManager.InstallByMsiException: The InstalledByMSI element in extension NuGet Package Project cannot be 'true' when installing an extension through the Extensions and Updates Installer.  The element can only be 'true' when an MSI lays down the extension manifest file.
   at Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService.CheckForInstallBlockers(InstallableExtensionImpl extension, IInstalledExtensionList modifiedInstalledExtensionsList, Boolean isNestedExtension, Boolean& olderVersionInstalled)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService.InstallInternal(InstallableExtensionImpl extension, Boolean perMachine, Boolean isNestedExtension, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, UInt64 totalBytesToWrite, UInt64& totalBytesWritten, Boolean enforceCertificateCheckForUpgrade)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService.BeginInstall(IInstallableExtension installableExtension, Boolean perMachine, AsyncOperation asyncOp, Boolean enforceCertificateCheckForUpgrade)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService.InstallWorker(IInstallableExtension extension, Boolean perMachine, AsyncOperation asyncOp, Boolean enforceCertificateCheckForUpgrade)
```

**Comment by petero-dk** [comment]
As far as I can tell, then you still have set 

```
InstalledByMsi="true"
```

So the vsix wont actually install by themselves?

**Comment by kovalikp** [comment]
Main issue is that NuProj relies on writing build system into `c:\Program Files (x86)\MSBuild\NuProj\`. I suspect VS doesn't allow extensions to write outside extensions directory. Hence the MSI installer.

**Comment by AArnott** [comment]
I already solved the distinct naming problem -- but I haven't uploaded the latest VSIXes.
As for the VSIXes being unwilling to install, that figures. I don't ever use the build authoring supplied by the MSI -- I always use the NuProj NuGet package for that, so the VSIX could theoretically be enough.
Would they be enough for you @ppittle (assuming they were willing to install) or do you actually need the MSI?

**Comment by petero-dk** [comment]
@AArnott on the topic of the nuget packages I cannot seem to see them in the feed you showed earlier, some other packages, but not the NuProj or NuProj.Common?

**Comment by AArnott** [comment]
@petero-dk are you showing prerelease packages? (that's often not shown by default).

**Comment by AArnott** [comment]
BTW yesterday I updated [our build doc](https://github.com/nuproj/nuproj/blob/master/docs/Build.md) to describe how to use the NuProj nuget package instead of the installed MSI.

**Comment by petero-dk** [comment]
@AArnott great build doc, I just have one small addition:

Using this for the property group would allow for the project to switch between build server and local (I could not load the project after changing the project file)
```
  <PropertyGroup Condition="'$(NuGetPackageRoot)' == ''">
    <NuGetPackageRoot>$(UserProfile)\.nuget\packages\</NuGetPackageRoot>
    <NuProjPath>$(NuGetPackageRoot)NuProj\0.10.48-beta-gea4a31bbc5\tools\</NuProjPath>
  </PropertyGroup>
  <PropertyGroup>
    <NuProjPath Condition="!Exists('$(NuProjPath)\NuProj.props')">$(MSBuildExtensionsPath)\NuProj\</NuProjPath>
  </PropertyGroup>
```

**Comment by AArnott** [comment]
Thanks, @petero-dk. I *did* have a bug in my docs where I put the condition in the wrong place and have fixed that. But yes, one fallout of this is that a package restore must happen before opening the project in VS. True, your change alleviates that, if the MSI happens to be installed. But as the MSI for customers is often the *very* old 0.10.4 version, which doesn't support project.json, I'm not sure that will tend to work well for the various scenarios where folks would open the project without a restore first.

**Comment by ppittle** [comment]
FYI, I reached out to MyGet support about uploading MSI installers along-side the VSIX:

> Hi,
> 
> Unfortunately this is not possible at the moment. Our VSIX feed is atom based, and I believe the MSI   downloads use a proprietary protocol from Microsoft.
>  
> Could the MSI be embedded in a VSIX perhaps? That could make things work.
>  
> Best regards,
> Maarten

So doesn't look like MyGet will support the installer as is.  Not sure what the stroy is creating a separate VSIX to package the original and the MSI.  But wanted to pass that along.

**Comment by AArnott** [comment]
OK, I just released 0.11.12-beta to the VS Gallery (thanks @terrajobst for enabling that!).
I will update the one on nuget.org as soon as I get publish rights to do that as well. In the meantime, you can get the matching nuget package from [my myget.org feed](https://www.myget.org/F/aarnott/api/v3/index.json)

**Comment by AArnott** [comment]
0.11.14-beta is now on the VS gallery and nuget.org gallery.
